init.lua

```diff
@@ -12,46 +12,26 @@
 ]]--
 
 
-
-local wp = minetest.get_worldpath() .. "/luscious"
-minetest.mkdir(wp)
-
 local mgp = minetest.get_mapgen_params()
 local chunksize = 16 * mgp.chunksize
 
-local function cmpy(p2, y)
-	if y > 0 then
-		local h1 = p2 % 16
-		local h2 = math.floor(p2 / 16) * 16
-
-		h1 = h1 - math.max(math.min(h1, math.floor(y / 16)), 0)
-		return math.max(1, h1 + h2)
-	else
-		return math.max(1, p2)
-	end
+local function chose_color(pos)
+    -- Calculate color from biome heat and humidity.
+    local biome_data = minetest.get_biome_data(pos)
+    local hum1 = biome_data.humidity
+    local temp1 = biome_data.heat
+    local vh = math.floor(math.min(math.max(math.floor(hum1), 0), 100) / 6.6)
+	local vt = math.floor(math.min(math.max(math.floor(temp1), 0), 100) / 6.6)
+	local c1 = (vt + (vh * 16))
+    --minetest.chat_send_all(temp1)
+    return c1
 end
-local function on_construct(pos)
-	-- get chunk from pos
-	local v = vector.apply(pos, function(a) return math.floor((a - 48) / chunksize) end)
-	local o = vector.subtract(pos, vector.apply(v, function(a) return (a * chunksize) + 48 end))
-	local l = o.z * (chunksize) + o.x
-	local p = minetest.hash_node_position(v)
-
-	local f = io.open(wp .. "/" .. string.format("%d", p), "r")
-	if not f then
-		minetest.log("error", "unable to find map for " .. string.format("%d", p))
-		return
-	end
 
-	local z = f:read("*a")
-	f:close()
-	local map = minetest.decompress(z)
 
-	local node = minetest.get_node(pos)
-	
-    if node.param2 ~=1 then     -- Use/preseve 1 as the flag to prevent leaf decay.     
-        node.param2 = cmpy(string.byte(map, l + 1), pos.y)
-    end
```

Comment:
You're throwing out all the caching here. It would be likely good for performance if you would still make the cache and use it instead of just calling the API function, since caching is cheap and very efficient here (since every vertical Y column has only a 16x16 cache block).

Reply:
Yes , I kinda overlooked the performance side I thought it was principllay meant to store the information for later use.  
I wonder how much slower it is.  Is there a built in benchmark for generation  compare the two?